### ISSUE TITLE
Setup page config functions not functional in 1.0.3/4 that worked in 1.02

### ISSUE THREAD

**Issue by micgresham** [cotestatnt/async-esp-fs-webserver#18]
Hi,

Several /setup page options that were working in 1.0.2 now do nothing in 1.03 or 1.04.  

-Custom logo not working
-Custom /setup page title not working
-Custom firmware version not working

See attached screen captures.

Here is the code:
// Web Server library
#include <AsyncFsWebServer.h>

// WebServer
AsyncFsWebServer asyncServer(WSPort,FILESYSTEM);

  // Add a new config box
  asyncServer.addOptionBox("Things");
  asyncServer.addOption("Build Date (MMDDYY)", buildDate);

   // Add custom page title to /setup
  char pageTitle[30];
  sprintf(pageTitle,"Setup %s",macID);
  asyncServer.setSetupPageTitle(pageTitle);
  
  // Add custom logo to /setup page with custom size:w
  asyncServer.setLogoBase64(base64_logo, "128", "128", /*overwite file*/ true);

  asyncServer.setFirmwareVersion(BUILD_VER);

  // Start webServer
  asyncServer.init();

<img width="1028" alt="v1 0 2" src="https://github.com/cotestatnt/async-esp-fs-webserver/assets/39626734/404c84e5-091a-4b54-8541-fd9b9f1f6d6c"> v1.0.2
<img width="1057" alt="v1 0 4" src="https://github.com/cotestatnt/async-esp-fs-webserver/assets/39626734/b3e86cd4-941f-4f81-88df-913774bc9ed0"> v1.0.3/1.0.4

- I noticed that the Options tab that was auto generating under 1.0.2 has vanished.  Attempts at creating the tab have failed.

Sorry for all of the issues.

-michael

**Comment by micgresham** [comment]
I found that if I manually add the param-box# to the config.json, it will show up.
{
  "param-box1": "Options",
  "IPaddr": "",
  "GW": "",
  "Mask": "",
  "DNS": "",
  "NWConfig": 0,
  "sensorType": 33,
  "BuildDate": "123123",
  "Backlight": true,
  "BacklightOffTime": 0,
  "BacklightOffDelay": 600000,
  "Feature": 0,
  "Category": 33,
  "WSPort": 80,
  "Units": 0,
  "TempOffset": 0,
  "TempMin": 0,
  "TempMax": 0,
  "HumidOffset": 0,
  "HumidMin": 0,
  "HumidMax": 0,
  "TempDev": 0,
  "WatchDog": true,
  "Hostname": "bite_me",
  "Server": "192.168.0.94:8090",
  "name-logo": "Setup 27FC6C123123",
  "img-logo": "/config/img-logo-128_128.txt",
  "param-box2": "Bite me!"

<img width="1048" alt="Untitled3" src="https://github.com/cotestatnt/async-esp-fs-webserver/assets/39626734/9d15985e-0f80-4bcc-8f9f-412130d9ce43">

So it looks like the function call to set the value  (asyncServer.addOptionBox) is not connecting....

I would also like a handler to intercept the config save.  When I do a pull down box, it is saving all of the options, not just the one I selected.  This creates a very large json file when the pulldown box is for timezones.

I will continue testing.

-michael

**Comment by cotestatnt** [comment]
Hi Michael.
To introduce the latest features (manual wifi configuration) I had to slightly modify the structure of the config.json file.
I tried to make the change backwards compatible, but I would say it didn't work. I'm sorry for the inconvenience.

To fix it, the advice is to delete or rename the current `config.json` file and restart the ESP so that a new one will be created.

**Comment by micgresham** [comment]
I will test this weekend.  What about a method of only showing the config elements I want to show on the Config page in /setup?

**Comment by cotestatnt** [comment]
Sorry, I do not understand what you mean.

**Comment by micgresham** [comment]
Everything in my config file, shows up on the config tab.  I would like to selectively display some config elements and not others.  See the attached image,

 
<img width="979" alt="setup-options" src="https://github.com/cotestatnt/async-esp-fs-webserver/assets/39626734/14137f10-9d09-4297-903b-0223b732dac9">

**Comment by micgresham** [comment]
>>To fix it, the advice is to delete or rename the current config.json file and restart the ESP so that a new one will be created.
Tried it.  Still no joy.

On another note: If I change the web server port from 80 to anything else on the 1.0.2 branch, the Options tab also disapears.

**Comment by micgresham** [comment]
I am still having problems with the calls to customize the HTML pages.

Here is code:

asyncServer = new AsyncFsWebServer(WSPort,FILESYSTEM);

  asyncServer->enableFsCodeEditor();
  asyncServer->setFsInfoCallback(getFsInfo);

  asyncServer->on("/", handleRoot);
  asyncServer->on("/stats", getStats);
  asyncServer->on("/stats/json", getStatsJSON);
  

  asyncServer->on("/connect", HTTP_POST, MydoWifiConnection);

  // Serve the /<macID>&Stats/ page
  asyncServer->on(("/" + String(macID) + "&Stats/").c_str(), HTTP_GET, getStats);

  // Serve the /<macID>&Stats/json page
  asyncServer->on(("/" + String(macID) + "&Stats/json").c_str(), HTTP_GET, getStatsJSON);

  // Add a new config box
  asyncServer->addOptionBox("Configuration");

 // asyncServer->addDropdownList("Time Zone", tzNames, N_ELEMENTS(tzNames));
  asyncServer->addOption("Backlight",Backlight,false);
  asyncServer->addOption("BacklightOffDelay",BacklightOffDelay,false);

   // Add custom page title to /setup
  char pageTitle[50];
  sprintf(pageTitle,"Watchman %s",macID);
  Serial.println(pageTitle);
  asyncServer->setSetupPageTitle(pageTitle);

Things like the page title contain a MAC address.  I have verified the pageTitle variable contains the full text; however, when I call the setSetupPageTitle, I am only getting up to the first space with no mac address.

Other things still do not display like the Configuration tab items.

Any suggestions?

<img width="1124" alt="screenshot#1" src="https://github.com/cotestatnt/async-esp-fs-webserver/assets/39626734/adaf23c9-1c55-45f9-bc1a-2c8723e22b8c">
<img width="1126" alt="screenshot#2" src="https://github.com/cotestatnt/async-esp-fs-webserver/assets/39626734/8a721710-3c36-404a-8799-8361635eed02">

**Comment by micgresham** [comment]
I think I found the problem.  I am using config.json for the application data and your library is also using it.  If I move my config to another named json file, the web page functions start working.  I also found that if I do not enable the edit page, I can provide my own handler for the POST of the /edit command.  Sadly, this does not work if the edit pages are enabled.

I would like to request that a option to provide an additional function to be called when the "Save Config" button is pressed.
